tests/apollo/test_skvbc_restart_recovery.py

```diff
@@ -549,6 +549,61 @@ async def test_view_change_with_non_primary_replica_in_state_transfer(self, bft_
                 run_ops=lambda: skvbc.run_concurrent_ops(num_ops=20, write_weight=1), threshold=20)
             log.log_message("fast path prevailed")
 
+    @skip_for_tls
+    @with_trio
+    @with_bft_network(start_replica_cmd, selected_configs=lambda n, f, c: c == 0, rotate_keys=True)
+    @verify_linearizability()
+    async def test_view_change_with_isolated_replicas(self, bft_network, tracker):
+        """
+        test View Changes with multiple View increments, where the
+        isolated F-1 expected next primaries will not be able to step in as
+        primaries, but will activate the corresponding view for which it is
+        theirs turn to become Primary.
+
+        Step by step scenario:
+        1. Use a one way isolating adversary to isolate the F-1 replicas after the current primary in such a way that they cannot send messages to the peers, but can receive messages from them.
+        2. Stop the current primary.
+        3. Send Client requests to trigger a View Change.
+        4. Wait for the system to finish View Change. Note that multiple View increments will happen.
+        5. Drop the network adversary and verify Fast Commit Path is recovered in the system by introducing client requests.
+
+        We can perform this test in a loop multiple times.
+        """
+
+        # start replicas
+        [bft_network.start_replica(i) for i in bft_network.all_replicas()]
+
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network, tracker)
+        loop_count = 0
+        while (loop_count < 100):
+            loop_count = loop_count + 1
+
+            primary = await bft_network.get_current_primary()
+
+            replicas_to_isolate = random.sample(
+                bft_network.all_replicas(without={primary}), bft_network.config.f - 1)
+
+            other_replicas = [primary]
+            for r1 in bft_network.all_replicas(without={primary}):
+                append = True
+                for r2 in replicas_to_isolate:
+                    if (r1 == r2):
+                        append = False
+                        break
+                if (append):
+                    other_replicas.append(r1)
+
+            view = await bft_network.get_current_view()
+
+            with net.ReplicaOneWayTwoSubsetsIsolatingAdversary(bft_network, [], replicas_to_isolate) as adversary:
```

Comment:
I think the ReplicaOneWayTwoSubsetsIsolatingAdversary takes as parameters 2 subsets - the first being the blocked receivers and the second - the blocked senders. Why is the set of blocked receivers empty here?

Reply:
"Use a one way isolating adversary to isolate the F-1 replicas after the current primary in such a way that they cannot send messages to the peers, but can receive messages from them."

Reply:
@HristoStaykov do you agree with @titusen's approach here?

Reply:
Our goal of using the **ReplicaOneWayTwoSubsetsIsolatingAdversary** is to make it impossible for the next F primaries to step in as such, therefore we should get 1 View Change with F view increments. We want them to be able to receive messages from the peers, but be unable to send to the peers. In this way neither of the F next primaries will be able to send his NewView message to the peers.

Reply:
So is it correct now? If no, what's wrong?

Reply:
Resolving this conversation, because the matter has been resolved vis zoom.